### PR TITLE
Fix srcset parse error

### PR DESCRIPTION
**Pull request by fisker**
fixes: #7234

- [stable playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeADgSzDArgJzgD4AdKAAkvNQGcICw5yb8wa4YBeEkAegFs4AE0wBDXnAAeo-ugA2cGrxoEAZnHwBaAEwAWAAw79+gHQArdAHMeZKnfuCRo7iAAU-TFE0B3TEJgAFkjkABzG6JIAlDykFFSomPyWzKwuAsJiEtKyCkroop7wQpoBolDF2gCcIZoAzLXa5lY85LyxqLxYOATEZCAANCAQ6DCY0DTIoKL4+BDeAArTCBMoogBuEH4DIABG+KJgANYcAMr5YJ6WyDD4uHCDATD8cgDqAZjwNOdwJ8sfmGsPgBPZDgGgTQaedj4GDzfaWfiiZCqURydiDMw0SQAIX2R1OMjgABlPHBkaj0SBMZITpcFABFXAQeDktH3ED5fDQ0GPZ7bdD4QovPyBZBhQYCiDsF77dCggWKDRrMmDACOTPgcOGKxAohomigcGEwm2hHVmEIcNECKRSBRbMG7A811u7JodLgjOZZLtFPZMFEO2F-gCyG0gxuBTklwAwhB+IjQVBoCqQLh2AAVQMre2UtZ3ACS5QQMBOYEFIwAguUTjAgQpWewAL5NoA)
- [PR playground](https://deploy-preview-7295--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAeADgSzDArgJzgD4AdKAAkvNQGcICw5yb8wa4YBeEkAegFs4AE0wBDXnAAeo-ugA2cGrxoEAZnHwBaAEwAWAAw79+gHQArdAHMeZKnfuCRo7iAAU-TFE0B3TEJgAFkjkABzG6JIAlDykFFSomPyWzKwuAsJiEtKyCkroop7wQpoBolDF2gCcIZoAzLXa5lY85LyxqLxYOATEZCAANCAQ6DCY0DTIoKL4+BDeAArTCBMoogBuEH4DIABG+KJgANYcAMr5YJ6WyDD4uHCDATD8cgDqAZjwNOdwJ8sfmGsPgBPZDgGgTQaedj4GDzfaWfiiZCqURydiDMw0SQAIX2R1OMjgABlPHBkaj0SBMZITpcFABFXAQeDktH3ED5fDQ0GPZ7bdD4QovPyBZBhQYCiDsF77dCggWKDRrMmDACOTPgcOGKxAohomigcGEwm2hHVmEIcNECKRSBRbMG7A811u7JodLgjOZZLtFPZMFEO2F-gCyG0gxuBTklwAwhB+IjQVBoCqQLh2AAVQMre2UtZ3ACS5QQMBOYEFIwAguUTjAgQpWewAL5NoA)

Notice: tests I added is not really catching error, because the error throw from this line

https://github.com/prettier/prettier/blob/5e8a4a115a181273e4b57dce8fb78d82863a784c/src/language-html/syntax-attribute.js#L22

is catched by this line https://github.com/prettier/prettier/blob/26183e5be2c07eeb392b3a6b7822517cf6f43fc6/src/main/ast-to-doc.js#L117 .

I tried to fix this in https://github.com/prettier/prettier/pull/7296 , but seems harder than I thought.

To prove what I said, add this

```html
<img srcset="/assets/visual@0.5.png 1x 2x"/>
```

to test, test will still pass, but [playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAlgWwOYAIDOATmPnDALwA6IA9AIb6kz40Bu6+ArnQDYACABgB0AViEAHKHgCMAD1wAmWdRoA+EABoQEcTHTR8yUHUKEIAdwAKJhIZR1WEdABNNIAEaE6YANZkAyuLe6FLIMISccFoAFjCYPADq0ejw+EFgcP62KejsMACeyOCMbiGkhDCWXtiYdMgAZrykWgBW+LIAQl6+AXSYcAAyIXANTVEgbbL+Idg8cACKnBDwozzNIEGE5UWx8W7ihCEwCS4w0cgAHAJaBxCkCV7iRQdw5awjWgCOS-BVOnYgBgAWigcDgznBbkIcG+6GhVToNTqSEaa3GpEw6DCEXRMzmi2WIxRYy0MDo7hOzjOyAUpK86B4MwAwhBMLUilBoB8QJxSAAVcl2VHrViRACSUAhsH8YEOugAgpL-AU5qtSABfdVAA) throws error.

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-html/syntax-attribute.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ function printImgSrcset(value) {
   const hasH = srcset.some(src => src.h);
   const hasX = srcset.some(src => src.d);
 
-  if (hasW + hasH + hasX !== 1) {
+  if (hasW + hasH + hasX > 1) {
     throw new Error(`Mixed descriptor in srcset is not supported`);
```
Comment (evilebottnawi) [inline src/language-html/syntax-attribute.js]: Hm, why we use "`" here ?

Comment (fisker) [reply src/language-html/syntax-attribute.js]: Haha, fixing

Comment (fisker) [reply src/language-html/syntax-attribute.js]: Maybe we can do something in prettier? Fix `` ` `` to `"`

Comment (evilebottnawi) [reply src/language-html/syntax-attribute.js]: I think we need enable the eslint `quotes` rule to avoid that in future

Comment (fisker) [reply src/language-html/syntax-attribute.js]: Opened a issue https://github.com/prettier/prettier/issues/7301

**Comment by thorn0** [comment]
I don't think this needs a fix. It turned out to be a playground-only error. I'm sorry for creating a misleading issue.

**Comment by fisker** [comment]
@thorn0 this still need fix 

```html
<img srcset="
             /media/examples/surfer-240-200.jpg
">
```

Stable

No changes

This PR:

```html
<img srcset="/media/examples/surfer-240-200.jpg" />
```

**Pull request merged**